### PR TITLE
Add agentic tool adapter status controls

### LINKED ISSUE #680: OpenSearch-VL alignment: unify tool runtime for training, benchmark, and evaluation - Milestone 2: implement deterministic local tool adapters - Unit 2: Add timeout, cancellation, and failure-stage tests for every tool adapter.
## Executable Unit

Add timeout, cancellation, and failure-stage tests for every tool adapter.

## Parent Context

- Direction: `OpenSearch-VL alignment: unify tool runtime for training, benchmark, and evaluation`
- Plan/Milestone: `Milestone 2: implement deterministic local tool adapters`

## Definition Of Done

- The change is scoped to the smallest useful implementation slice.
- The governing spec or plan is created or updated before broad code changes.
- Tests cover the new behavior or contract edge case.
- Metrics and verification output are recorded in the PR evidence.

## Suggested Verification

- `git diff --check`
- Focused pytest or Swift test for the touched code path
- Changed-scope coverage for modified Python or Swift files when measurable
- Any relevant benchmark/evaluation artifact if this changes measured behavior

### PR DESCRIPTION
**Pull request by Zigfreidish**
## Summary
- Add fixture-driven status controls for the deterministic agentic tool runtime so every built-in adapter can exercise `timeout`, `failed`, and cancellation paths.
- Document the cancellation mapping as observation status `failed` with `failure_stage=cancelled` and `cancelled=true` payload evidence.
- Extend focused tests across `image_crop`, `layout_parse`, `text_search`, `image_search`, `visit`, and `local_compute`, including call-id precedence, wildcard controls, invalid controls, and the review-requested sequential override lookup.

Closes #680.
Refs #674, #678.

## Plan or Spec
- `docs/plans/2026-05-20-agentic-tool-adapter-status-controls.md`
- `docs/unified-agentic-tool-runtime-contract.md`

## Commands Run
```text
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_agentic_tools.py
# 29 passed in 0.04s

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" COVERAGE_FILE=/tmp/agentic-tool-status.coverage uv run --project services/mlx-worker-python coverage run -m pytest -q services/mlx-worker-python/tests/test_agentic_tools.py && PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" COVERAGE_FILE=/tmp/agentic-tool-status.coverage uv run --project services/mlx-worker-python coverage json -o /tmp/agentic-tool-status-coverage.json && PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --project services/mlx-worker-python python scripts/python_changed_line_coverage.py --coverage-json /tmp/agentic-tool-status-coverage.json --diff-from origin/main services/mlx-worker-python/worker/runtime/agentic_tools.py services/mlx-worker-python/tests/test_agentic_tools.py
# TOTAL 100.00% 55/55

git diff --check
# passed

git commit -m "Add agentic tool adapter status controls"
# pre-commit enforced full local gate on 256 GiB macOS host:
# make swift-test: rc=0, elapsed=597.7s
# make py-test: 2836 passed, 14 skipped, 2 warnings in 129.82s
# make integration-test: 114 passed, 1 skipped in 628.03s
# Melix PR Scoped Performance Report: Status ok; selected probes 0; regressions 0; verification failures 0

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_agentic_tools.py
# 29 passed in 0.04s

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" COVERAGE_FILE=/tmp/agentic-tool-status-review.coverage uv run --project services/mlx-worker-python coverage run -m pytest -q services/mlx-worker-python/tests/test_agentic_tools.py && PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" COVERAGE_FILE=/tmp/agentic-tool-status-review.coverage uv run --project services/mlx-worker-python coverage json -o /tmp/agentic-tool-status-review-coverage.json && PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --project services/mlx-worker-python python scripts/python_changed_line_coverage.py --coverage-json /tmp/agentic-tool-status-review-coverage.json --diff-from origin/main services/mlx-worker-python/worker/runtime/agentic_tools.py services/mlx-worker-python/tests/test_agentic_tools.py
# TOTAL 100.00% 59/59

git diff --check origin/main...HEAD
# passed

git commit -m "Address agentic status override review"
# pre-commit enforced full local gate on 256 GiB macOS host:
# make swift-test: rc=0, elapsed=178.7s
# make py-test: 2837 passed, 14 skipped, 2 warnings in 117.70s
# make integration-test: 114 passed, 1 skipped in 309.98s
# Melix PR Scoped Performance Report: Status ok; selected probes 0; regressions 0; verification failures 0
```

## Coverage and Metrics
- Changed-line coverage: 100.00% (59/59) for `services/mlx-worker-python/worker/runtime/agentic_tools.py` and `services/mlx-worker-python/tests/test_agentic_tools.py` after the review fix.
- Adapter status matrix: 6 built-in adapters x 3 status controls = 18 deterministic adapter status assertions for `timeout`, `failed`, and cancellation.
- Performance reports: `.runtime/pre-commit-performance/20260519-164039-0dc17925/report/report.md` and `.runtime/pre-commit-performance/20260519-173511-45faa07b/report/report.md`; both `Status: ok`, `Regressions: 0`, `Verification failures: 0`.
- Observability mode: evidence/test fixture only. Probe overhead: N/A because this adds deterministic fixture controls and documentation without production runtime metrics, PR-scoped probes, or protocol/dependency changes.

## Known Gaps
- Real asynchronous cancellation for external providers is out of scope; this change covers the deterministic local agentic tool runtime fixture path needed for #680.
- No protocol, generated artifact, lockfile, or dependency changes.

## Evidence Checklist
- [x] Relevant plan or spec is identified.
- [x] Behavior changes are reflected in the relevant docs.
- [x] Protocol changes include regenerated generated artifacts. N/A: no protocol changes.
- [x] Dependency changes include updated lockfiles. N/A: no dependency changes.
- [x] Relevant tests were run.
- [x] A metrics report is included, or `N/A` is stated explicitly with the reason.
- [x] Observability mode, probe overhead, and any deferred debug or evidence work are recorded, or `N/A` is stated explicitly with the reason.
- [x] Deferred work and known gaps are stated explicitly.

### CODE REVIEW

File: services/mlx-worker-python/worker/runtime/agentic_tools.py [hunk 1]
```diff
@@ -209,6 +226,48 @@ def _validate_required_arguments(descriptor: ToolDescriptor, arguments: dict[str
...
 
 
+def _status_override_payload(
+    *,
+    fixture_context: dict[str, Any],
+    tool_name: str,
+    tool_call_id: str,
+) -> dict[str, Any] | None:
+    overrides = _context_mapping(fixture_context, "tool_status_overrides")
+    raw_override = overrides.get(tool_call_id, overrides.get(tool_name, overrides.get("*")))
```
[the remark this answers is not in the document]
Comment (Zigfreidish) [reply services/mlx-worker-python/worker/runtime/agentic_tools.py]: Applied in 01b9de52 by switching the override lookup to sequential checks, so the tool-name and wildcard lookups only run when the call-id override is absent. Re-ran focused tests and the full local pre-commit gate; all passed.
